python/cuml/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -44,19 +44,19 @@ from libc.stdint cimport int64_t, uint32_t, uintptr_t
 from libcpp cimport bool
 from libcpp.vector cimport vector
 from pylibraft.common.handle cimport handle_t
+from pylibraft.common.mdspan cimport *
```

Comment:
I'm aware that this a common pattern in this codebase, but should we try to avoid wildcard imports in the future?

Reply:
Can you explain why we need to avoid them? Happy to not do this, just want to know for my own knowledge.

Reply:
Quoting the ["imports" section from PEP 8](https://peps.python.org/pep-0008/#imports):

> Wildcard imports (from <module> import *) should be avoided, as they make it unclear which names are present in the namespace, confusing both readers and many automated tools. [...]

It obfuscates what's actually present in the namespace which makes it harder to understand what exact interface is exposed through the module and what a symbol's provenance is. This might be a Cython pattern that I am not aware of, but for general Python code this is an indisputed anti-pattern.

Reply:
I think it's less bad and not necessarily an antipattern for cimports. It's not uncommon in cython codebases to have large headers and include them with `cimport *` (using `cimport *` is basically the same as a C include). `pyarrow` does this a bunch, for example.

Qualified imports make it easier to understand what's being pulled in, and also lets linters check when a cimport is no longer needed (I just removed a bunch of unnecessary ones in #6600, for example). I wouldn't block on adding a `cimport *`, but if the number of included symbols is small, I also think it'd be nicer to spell them out explicitly.

Reply:
Definitely not blocking for this PR. Just wanted to get a take on this.